providers/azure/provider/provider.go

```diff
@@ -57,7 +57,10 @@ func (s *Service) ParseCLI(req *plugin.ParseCLIReq) (*plugin.ParseCLIRes, error)
 	if len(subscriptionsToExclude.Value) > 0 {
 		opts["subscriptions-exclude"] = string(subscriptionsToExclude.Value)
 	}
-
+	// the presence of the flag indicates that we should skip cleanup
+	if len(skipSnapshotCleanup.Value) > 0 {
```

Comment:
This might be counterintuitive.

Setting `--skip-snapshot-cleanup=false` will result in the option being true.